File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        char_set = set()
        left = 0
        max_score = 0

        for right in range(len(s)):

            while s[right] in char_set:
                char_set.remove(s[left])
                left += 1

            char_set.add(s[right])

            max_score = max(max_score, right-left+1)

        return max_score

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-0.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        int count[26] = {}; int maxf = 0; int l = 0;
        for (int r = 0; r < s.size(); r++){
            maxf = max(maxf, ++count[s[r] - 'A']);
            if (r - l + 1 - maxf > k ) count[s[l++] - 'A']--;
        }
        return s.size() - l;
    }
};
